core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java

```diff
@@ -344,6 +359,24 @@ private static int calculateMaxFill(int hashSize)
         return maxFill;
     }
 
+    private void updateDictionaryLookBack(Block dictionary)
+    {
+        if (dictionaryLookBack == null || dictionaryLookBack.getDictionary() != dictionary) {
+            dictionaryLookBack = new DictionaryLookBack(dictionary);
+        }
+    }
+
+    private int getGroupId(Block dictionary, int positionInDictionary)
```

Comment:
nit: can you name it in a way that it does suggest it has side effects `registerGroupId`?